tests/experimental/test_experimental_utils.py

```diff
@@ -0,0 +1,508 @@
+import numpy as np
+import pytest
+import torch
+from autoemulate.experimental.data.utils import (
+    ConversionMixin,
+    Standardizer,
+    ValidationMixin,
+)
+from autoemulate.experimental.types import NumpyLike
+from torch.utils.data import DataLoader, Subset, TensorDataset
+
+
+class TestConversionMixin:
+    """
+    Class to test the ConversionMixin class.
+    """
+
+    def setup_method(self):
+        """
+        Define the ConversionMixin instance.
+        """
+        self.mixin = ConversionMixin()
+
+    def test_convert_to_dataset_numpy(self):
+        """
+        Test converting numpy arrays to a Dataset.
+        """
+        X = np.array([[1.0, 2.0], [3.0, 4.0]])
+        y = np.array([1.0, 2.0])
+        dataset = self.mixin._convert_to_dataset(X, y)
+
+        # TODO: is the distinction between TensorDataset and Dataset important?
+        assert isinstance(dataset, TensorDataset)
+        assert torch.equal(dataset.tensors[0], torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
+        assert torch.equal(dataset.tensors[1], torch.tensor([1.0, 2.0]))
+
+    def test_convert_to_dataset_tensor(self):
+        """
+        Test converting torch tensors to a Dataset.
+        """
+        X = torch.tensor([[1.0, 2.0], [3.0, 4.0]])
+        y = torch.tensor([1.0, 2.0])
+        dataset = self.mixin._convert_to_dataset(X, y)
+
+        assert isinstance(dataset, TensorDataset)
+        assert torch.equal(dataset.tensors[0], X)
+        assert torch.equal(dataset.tensors[1], y)
+
+    def test_convert_to_dataset_invalid(self):
+        """
+        Test invalid input to _convert_to_dataset.
+        """
+        X = "invalid input"
+        with pytest.raises(ValueError, match="Unsupported type for x"):
+            self.mixin._convert_to_dataset(X)  # type: ignore - test for invalid type
+
+    def test_convert_to_tensors(self):
+        """
+        Test converting input data to tensors.
+        """
+        X = np.array([[1.0, 2.0], [3.0, 4.0]])
+        y = np.array([1.0, 2.0])
+        x_tensor, y_tensor = self.mixin._convert_to_tensors(X, y)
+
+        assert isinstance(x_tensor, torch.Tensor)
+        assert isinstance(y_tensor, torch.Tensor)
+        assert torch.equal(x_tensor, torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
+        assert torch.equal(y_tensor, torch.tensor([[1.0], [2.0]]))
+
+    def test_convert_to_tensors_subset(self):
+        """
+        Test converting a Subset of a TensorDataset to tensors.
+        """
+        X = torch.tensor([[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]])
+        y = torch.tensor([1.0, 2.0, 3.0])
+        dataset = TensorDataset(X, y)
+        subset = Subset(dataset, [0, 1])
+        x_tensor, y_tensor = self.mixin._convert_to_tensors(subset)
+
+        assert torch.equal(x_tensor, torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
+        assert torch.equal(y_tensor, torch.tensor([[1.0], [2.0]]))
+
+    # def test_convert_to_tensors_invalid(self):
```

Comment:
See [above comment](https://github.com/alan-turing-institute/autoemulate/pull/464#discussion_r2095306906) why this is commented out for now